admin/admin cleanup

dwaynebailey opened the pull request:
This PR:

1. Adds a check for admin/admin and warns if the default password is still set (this should catch bad setup on old installs)
2. Removes the creation of admin from initdb

I did look at docs cleanup but it moved so quickly from a simple lets check if we need ``createadminuser`` instructions to a cleanup of the upgrade page.  So lets leave that out of this PR please.

I don't want to automate the default admin user creation but rather make that part of the setup of the server.  Installation should be fines, its documented.  Upgrade should also be fine as you'll have an admin user, you might however not have one with an email address but we can deal with that in the upgrade docs.

julen commented on pootle/checks.py:
```diff
@@ -235,3 +235,27 @@ def check_settings(app_configs=None, **kwargs):
+    try:
+        admin_user = get_user_model().objects.get(username='admin')
+        if admin_user is not None:
+            admin_authenticated = authenticate(username='admin', password='admin')
```
No need to `authenticate()`, just do `if admin_user.check_password('admin'):`

dwaynebailey replied:
Nice

julen commented on pootle/checks.py:
```diff
@@ -235,3 +235,27 @@ def check_settings(app_configs=None, **kwargs):
+                    hint=_("Remove the 'admin' user or change its password."),
+                    id="pootle.W016",
+                ))
+    except:
```
Better:
```python
User = get_user_model()
try:
    admin_user = User.objects.get(username='admin')
except User.DoesNotExist:
    pass
else:
    if admin_user.check_password('admin'):
        errors.append(...)
```

dwaynebailey commented:
@julen @jleclanche addressed your comments.

I've landed docs changes now also.  And added the warning post initdb to assist seasoned Pootle users so that they don't get caught out and we're not answering questions on IRC.

jleclanche commented:
gtm

julen commented on docs/server/upgrading.rst:
```diff
@@ -179,6 +179,15 @@ To perform the upgrade follow the next steps:
 Post-upgrade adjustments
 ------------------------
 
+* Check that you can login as an admin user.  If you existing adin user has no
```
*your existing admin*

julen commented on docs/server/upgrading.rst:
```diff
@@ -179,6 +179,15 @@ To perform the upgrade follow the next steps:
 
+* Check that you can login as an admin user.  If you existing adin user has no
+  email address then you will not be able to login.  In order the fix this use
+  :djadmin:`createsuperuser` to create a new superusers, or to create a
```
*superuser* singular

julen commented:
Tests broke I'm afraid; good to go provided that's fixed.

unho commented on docs/server/upgrading.rst:
```diff
@@ -179,6 +179,15 @@ To perform the upgrade follow the next steps:
+
+  .. code-block:: bash
+
+     (env) $ pootle createsuperuser
```
This is not post-upgrade adjustment. It is a step of the upgrade, so this must be before "Post-upgrade adjustments" section in the upgrade instructions.

dwaynebailey replied:
I didn't want to fiddle those docs anymore.  I have some upgrade cleanups and will deal with that with those.

unho commented on docs/server/upgrading.rst:
```diff
@@ -179,6 +179,15 @@ To perform the upgrade follow the next steps:
 Post-upgrade adjustments
 ------------------------
 
+* Check that you can login as an admin user.  If your existing adin user has no
```
Typo: adin
